app/src/test/java/br/com/chucknorrisfacts/ViewModelTest.kt

```diff
@@ -0,0 +1,222 @@
+package br.com.chucknorrisfacts
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule
+import br.com.chucknorrisfacts.home.entry.SearchFactsEntryModel
+import br.com.chucknorrisfacts.home.handler.FactHandler
+import br.com.chucknorrisfacts.home.service.HomeService
+import br.com.chucknorrisfacts.home.status.State
+import br.com.chucknorrisfacts.home.viewmodel.HomeViewModel
+import br.com.chucknorrisfacts.webservice.WebServiceAPI
+import br.com.chucknorrisfacts.webservice.exceptions.*
+import br.com.chucknorrisfacts.webservice.payload.SearchList
+import br.com.chucknorrisfacts.webservice.payload.SearchPayload
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Observable
+import junit.framework.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations.initMocks
+
+/**
+ * @rodrigohsb
+ */
+class ViewModelTest {
+
+    @Rule
+    @JvmField
+    val rule = InstantTaskExecutorRule()
+
+    @Mock
+    private lateinit var homeService: HomeService
+
+    @Mock
+    private lateinit var factHandler: FactHandler
+
+    @Mock
+    private lateinit var webServiceAPI: WebServiceAPI
+
+    private lateinit var viewModel: HomeViewModel
+
+    @Before
+    fun setup() {
+        initMocks(this)
+        viewModel = HomeViewModel(homeService, factHandler)
+    }
+
+    private fun createPayload(): SearchPayload {
+
+        val categories = ArrayList<String>()
+        categories.add("")
+        categories.add("")
+        categories.add("")
+        categories.add("")
+        val search = SearchList("", "", "", "", categories)
+
+        val searchList = ArrayList<SearchList>()
+        searchList.add(search)
+        searchList.add(search)
+        searchList.add(search)
+        searchList.add(search)
+        searchList.add(search)
+
+        return SearchPayload(5, searchList)
+
+    }
+
+    @Test
+    fun `test if status is waiting for input`() {
+
+        val testObserver = viewModel.fetchState().test()
+
+        testObserver.awaitTerminalEvent()
+
+        testObserver
+        .assertNoErrors()
+        .assertValue { status -> status is State.WaitingForInput }
+        .isTerminated
+    }
+
+    @Test
+    fun `test if status is Loading and then has content`() {
+
+        val searchFactsEntryModel = SearchFactsEntryModel(1, "", emptyList())
+
+        whenever(homeService.searchFacts("")).thenReturn(Observable.just(createPayload()))
+        whenever(factHandler.convert(createPayload(), "")).thenReturn(searchFactsEntryModel)
+
+        val testObserver = viewModel.searchContent("").test()
+
+        testObserver.awaitTerminalEvent()
+
+        val values = testObserver
+                .assertComplete()
+                .assertNoErrors()
+                .values()
+
+        assertTrue((values.size == 2)
```

Comment:
Não gostei dessa forma não mas foi o único jeito que consegui...
tava tentando assim mas não sei pq não funcionou:

testObserver
        .assertComplete()
        .assertNoErrors()
        .assertValue { status -> status is State.Success }
        .isTerminated

O lado bom de como ficou é que eu posso testar se o "Loading" veio primeiro.